cdap-security/src/main/java/io/cdap/cdap/security/store/file/SecureStoreDataCodecV1.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.store.file;
+
+import com.google.common.base.Preconditions;
+import io.cdap.cdap.api.security.store.SecureStoreData;
+import io.cdap.cdap.api.security.store.SecureStoreMetadata;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Version 1 codec for {@link SecureStoreData} and providing the {@link java.security.KeyStore} scheme.
+ *
+ * Version 1 codec uses JCEKS KeyStore.
+ *
+ * NOTE: This codec should not be used in normal operation and remains here for backwards compatibility purposes.
+ *       See {@link SecureStoreDataCodecV2} instead.
+ */
+@Deprecated
+public class SecureStoreDataCodecV1 implements FileSecureStoreCodec {
```

Comment:
Can we document the encoding format in the javadoc so that one doesn't need to look at the code to have an understanding? Same for the V2 codec.

Reply:
Done